### ISSUE TITLE
When adding pages to nav it uses pageid in url

### ISSUE THREAD

**Issue by silverelizard** [pencilblue/pencilblue#615]
To use the SEO and user-friendly slug style links defined on our pages, you must put the nav in yourself. If you use the page association tool in the nav, it automatically uses the id, instead.

**Comment by brianhyder** [comment]
The game plan is to tackle this when the nav service is moved over to the base object service implementation.  When ever a page or article is changed the cache site-map potentially needs to update.

**Comment by PieturP** [comment]
Is there any timeframe on when this issue will be tackled? — Thanks in advance for your answer :)

**Comment by brianhyder** [comment]
Hey @PieturP,

  I've been putting it off because I've been trying to think of a good way to implement this.  We could easily change it over to use the slug but in the event that the slug changed the nav item would have to be manually updated.  

We just didn't want to have to place this burden on our users.  

Ideally, what we would like to do is to provide an API that allows for easily manipulating the navigation structure and allow for n-tiered navigation.  This requires the re-work of the back end and requires a larger effort.  One of my biggest gripes now is the in-efficiency of the model we have in place.  

Maybe it is time to bite the bullet and knock it out.

What I'd like to know from our user base is what is more important to them.  This or the Localization items on the issues list?

@silverelizard  @PieturP thoughts?

**Comment by PieturP** [comment]
I see your point. Personally, I'd like to think that user friendliness and SEO are more important than possible dead-links when users change slugs — but that might just be me.
Ideally one could indeed expect the URL's in the navigation to change whenever the slug changes.

On the other hand, if such URL's were to be indexed by search engines or referenced from other pages, they would of-course also end up in 404's whenever they change. So it maybe a better choice to alert or warn people before changing slugs in general; e.g. have a confirm prompt in the admin interface or something.

**Comment by v3rt1go** [comment]
I agree with PieturP on this. SEO is one of the deciding factors for most of the clients.
I've found a workaround for this, by adding Link as the nav element and manually putting the slug /page/page-name for the link, it has pretty much the behavior I wanted to achieve.

**Comment by jobe451** [comment]
I tried this workaround. Linking to a page via /page/page-name does not bring up the requested page. Only /page/pageId get's me there. I would like to link to pages from wysiwyg-content (from other pages / articles). This currently is cumbersome, especially since I don't want to have all pages registered in the navigation .

**Comment by jobe451** [comment]
Please ignore my previous comment. /page/page-name  was not working for me, because I had a modified index-controller be in charge of the /page route. Because I did not want to use the article template for displaying page content.

**Comment by HadenHiles** [comment]
Will this navigation/slug enhancement be included in the 5.0.0 release? Thanks in advance.

**Comment by brianhyder** [comment]
@HadenHiles No, it is not.  Refactoring the way the navigation works is slated for sometime in 2016.  We will be working out the new year's roadmap the first couple of weeks in January.

**Comment by HadenHiles** [comment]
Thanks @brianhyder

**Comment by brianhyder** [comment]
In design:
https://github.com/pencilblue/pencilblue/wiki/Site-Navigation
